src/target/riscv/batch.c

```diff
@@ -135,7 +160,10 @@ int riscv_batch_run_from(struct riscv_batch *batch, size_t start_idx,
 
 		const bool delays_were_reset = resets_delays
 			&& (i >= reset_delays_after);
-		if (idle_count > 0 && !delays_were_reset)
+		const int idle_count = get_idle_count(batch, i, idle_counts);
+
+		assert(idle_count >= 0);
+		if (!delays_were_reset)
 			jtag_add_runtest(idle_count, TAP_IDLE);
```

Comment:
As noted in a previous comment, please prefer to use an unsigned type for the idle counts. 

The fact that jtag_add_runtest() uses a singed parameter looks like a mistake/oversight to me (it is 15 years old code).

```suggestion
		const unsigned int idle_count = get_idle_count(batch, i, idle_counts);

		if (!delays_were_reset)
			jtag_add_runtest((int)idle_count, TAP_IDLE);
```

Reply:
Addressed. Although here it's still an `int` to protect against integer overflow.